File: Data Structures & Algorithms/non-cyclical-number/submission-0.py
class Solution:
    def sumofDigits(self, n)->int:
        sum = 0
        while(n):
            sum += (n%10)*(n%10)
            n = n//10
        return sum
    def isHappy(self, n: int) -> bool:
        seen = list()
        while(n!=1):
            n = self.sumofDigits(n)
            if (n in seen):
                return False
            else:
                seen.append(n)
        return True
        

File: Data Structures & Algorithms/plus-one/submission-0.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        // Iterate from the last and add 1 until you have carry as 0
        int right = digits.size() - 1;
        int carry = 0;
        do {
            int sum = digits[right] + 1;
            carry = sum/10;
            digits[right] = sum%10;
            right--;
        } while((carry == 1)&& right>=0);

        // If carry is 1, insert 1 at the begining. 
        if (carry)
           digits.insert(digits.begin()+0, 1);

        return digits;
    }
};
